### PR TITLE
BAEL-7410: Finding Parent of a Node in a Binary Tree

### PR DESCRIPTION
**Pull request by pedrolopes9-7**

### CODE REVIEW

**Review by JavierPeris — CHANGES_REQUESTED** [review]
Please apply Baeldung formatter.

File: algorithms-modules/algorithms-miscellaneous-7/src/main/java/com/baeldung/algorithms/parentnodebinaytree/TreeNode.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            }
+
+            current = current.right;
+        }
+
+        throw new NoSuchElementException(format("No parent node found for 'target.value=%s' " +
+                        "The target is not in the tree or the target is the topmost root node.",
+                target.value));
+    }
+}
```
[the remark this answers is not in the document]
Comment (pedrolopes9-7) [reply algorithms-modules/algorithms-miscellaneous-7/src/main/java/com/baeldung/algorithms/parentnodebinaytree/TreeNode.java:114]: That makes sense, thanks. Will apply

File: algorithms-modules/algorithms-miscellaneous-7/src/main/java/com/baeldung/algorithms/parentnodebinaytree/TreeNode.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        if (isTargetChildOfCurrent) {
+            return current;
+        }
+
+        if (target.value < current.value) {
+            return parent(current.left, target);
+        }
+
+        return parent(current.right, target);
+    }
```
[the remark this answers is not in the document]
Comment (pedrolopes9-7) [reply algorithms-modules/algorithms-miscellaneous-7/src/main/java/com/baeldung/algorithms/parentnodebinaytree/TreeNode.java:82]: Sure, I´ve used variables for more meaningful names about what´s happening. But, I think we can explain the if statement in the text

**Comment by pedrolopes9-7** [comment]
Hey @JavierPeris, I've made your suggested changes here. Let me know if you agree with the code now, so I can update the article draft.
Thanks!

**Pull request merged**